### PR TITLE
Add Docs for Azure OpenAI fine-tuning

### PR DESCRIPTION
**Pull request by ash0ts**
## Description

What does the pull request do? If it fixes a bug or resolves a feature request, be sure to link to that issue.

## Ticket

Does this PR fix an existing issue? If yes, provide a link to the ticket here:

## Checklist

Check if your PR fulfills the following requirements. Put an `X` in the boxes that apply.

- [X] Files I edited were previewed on my local development server with `yarn start`. My changes did not break the local preview.
- [X] Build (`yarn docusaurus build`) was run locally and successfully without errors or warnings.
- [X] I merged the latest changes from `main` into my feature branch before submitting this PR.

### CODE REVIEW

**Comment by ash0ts** [comment]
@ngrayluna How do I get this new docs page to show up on the sidebar? It didn't show up when I ran it locally but the page rendered when directly typing in the URL

**Comment by ngrayluna** [comment]
Ah. Yes, you'll want to add  a new line to the sidebar.js file as well. :) 
Screen Shot 2024-01-09 at 10 47 24 AM

**Comment by ash0ts** [comment]
Thanks @ngrayluna! Ready for review :)

**Comment by ash0ts** [comment]
```
E100 [vocab] Runtime error

'config/vocabularies/Docs' directory does not exist
```

Not sure what this is/where this error comes from

**Comment by morganmcg1** [comment]
@ash0ts one nit, can you re-naming the files to lead with "azure...", we should consider this an Azure service, not a OpenAi service. There'll likely be a Azure ML section coming at some point too, so setting a good precedent here would be good

File: docs/guides/integrations/other/openai-azure-fine-tuning.md [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+
+# Azure OpenAI Fine-Tuning
+
+## Introduction
+Fine-tuning GPT-3.5 or GPT-4 models on Microsoft Azure using Weights & Biases allows for detailed tracking and analysis of model performance. This guide extends the concepts from the [OpenAI Fine-Tuning guide](/guides/integrations/openai) with specific steps and features for Azure OpenAI.
+
+![](/images/integrations/open_ai_auto_scan.png)
+
+:::info
+The Weights and Biases fine-tuning integration works with `openai >= 1.0`. Please install the latest version of `openai` by doing `pip install -U openai`.
```
Comment (morganmcg1) [inline docs/guides/integrations/other/openai-azure-fine-tuning.md]: I don't think this is needed, the 1.0 release is rapidly fading into the past :)

Comment (morganmcg1) [reply docs/guides/integrations/other/openai-azure-fine-tuning.md]: ah I see its in our OpenAI FT docs too, could you remove it from there also?

**Comment by morganmcg1** [comment]
Looking at this, lets just make this page more like a "diff" of the OpenAI page: https://docs.wandb.ai/guides/integrations/openai

Assuming nothing about the api is different, just show people the changes needed for W&B fine-tuning with Azure vs OpenAI. And direct people towards the OpenAI for everything else that isn't different. That way we only have to maintain 1 page when there are openai api updates etc.

**Review by morganmcg1 — DISMISSED** [review]
Lets re-write this to just be a short "diff" page of Azure OpenAI vs OpenAI. Let more detail in the comment

File: docs/guides/integrations/other/openai-azure-fine-tuning.md (hunk 1 above)
Comment (ash0ts) [reply docs/guides/integrations/other/openai-azure-fine-tuning.md]: I added as for some reason in the official Microsoft documentation still defaults to 0.28.1. This confused our external author. Can remove though!

**Comment by ash0ts** [comment]
> Looking at this, lets just make this page more like a "diff" of the OpenAI page: https://docs.wandb.ai/guides/integrations/openai
> 
> Assuming nothing about the api is different, just show people the changes needed for W&B fine-tuning with Azure vs OpenAI. And direct people towards the OpenAI for everything else that isn't different. That way we only have to maintain 1 page when there are openai api updates etc.

Transparently more detail was added to give some "meat" to the docs for the Microsoft folks. Realistically the only thing we really need is a single tab in the original OpenAI docs to toggle between "OpenAI" client vs. "AzureOpenAI". Outside the comment posted is there anything else that is considered too much?

**Comment by ngrayluna** [comment]
> ```
> E100 [vocab] Runtime error
> 
> 'config/vocabularies/Docs' directory does not exist
> ```
> 
> Not sure what this is/where this error comes from

Ignore this. Our doc linter temporarily broke. But it's fixed now :)

**Review by ngrayluna — APPROVED** [review]

**Pull request merged**